cvat/apps/dashboard/views.py

```diff
@@ -97,22 +97,39 @@ def DetailTaskInfo(request, task, dst_dict):
 @login_required
 @permission_required('engine.view_task', raise_exception=True)
 def DashboardView(request):
-    filter_name = request.GET['search'] if 'search' in request.GET else None
-    tasks_query_set = list(TaskModel.objects.prefetch_related('segment_set').order_by('-created_date').all())
-    if filter_name is not None:
-        tasks_query_set = list(filter(lambda x: filter_name.lower() in x.name.lower(), tasks_query_set))
+    query_name = request.GET['search'] if 'search' in request.GET else None
+    query_job = int(request.GET['jid']) if 'jid' in request.GET and request.GET['jid'].isdigit() else None
+    query_page = None
+    elements_per_page = 20
+    task_list = None
+
+    if query_job is not None and JobModel.objects.filter(pk = query_job).exists():
+        task_list = [JobModel.objects.select_related('segment__task').get(pk = query_job).segment.task]
+    else:
+        task_list = list(TaskModel.objects.prefetch_related('segment_set').order_by('-created_date').all())
+        if query_name is not None:
+            task_list = list(filter(lambda x: query_name.lower() in x.name.lower(), task_list))
+        query_page = int(request.GET['page']) if 'page' in request.GET and request.GET['page'].isdigit() else None
+
+    if query_page is not None:
+        start, stop = (query_page - 1) * elements_per_page, query_page * elements_per_page - 1    # 0, 19; 20, 39, 40, 59 etc
+        task_list = [ None if idx < start or idx > stop else task for idx, task in enumerate(task_list) ]
 
     data = []
-    for task in tasks_query_set:
-        task_info = {}
-        MainTaskInfo(task, task_info)
-        DetailTaskInfo(request, task, task_info)
-        data.append(task_info)
+    for task in task_list:
+        if task is not None:
+            task_info = {}
+            MainTaskInfo(task, task_info)
```

Comment:
Let's try to keep original `task` object without converting it to a dictionary. If you need to have a special method you can add them to `task` directly. Thus you don't need to have none. Pagination will not evaluate the object if it isn't necessary.